### PR TITLE
Feature/simplify configuration after port fix

### PR DESCRIPTION
**Pull request by marcules**
**Description of the change**

After comparing `deployment.yaml` with `statefulset.yaml` and `pvc.yaml` with `statefulset.yaml` I've seen that there are differences introduced in a regression when the sts was first added and deployment.yaml was first removed and added in again - part of this fix normalizes both configurations.

Additionally, now, that the port-validation fix is introduced in upstream uptime-kuma a exported service in the environment variables would not cause the same issue with invalid port numbers anymore - therefore these changes were simplified and moved into helper, as well as using the pod/container port name instead of the value, making it possible to configure the exposed port in the container image through `values.yaml` and also set a different port in the service through values.yaml as well.

**Benefits**

Make the service easier to configure and make it so deployment+pvc and statefulset behave the same.

**Possible drawbacks**

n/a

**Applicable issues**

n/a

**Additional information**

Variables were not updated, I think this happens automatically now, right? @dirsigler - if that is the case, maybe that step can be removed from the PR-template?

**Checklist**

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by dirsigler** [comment]
@marcules Thank you very much for your changes.
It is true, the whole deployment vs. statefulset topic added a bit of overhead which affected the overall Chart quality.

Value changes are added when the pre-commit framework is installed and used.
Means:
```
# Install framework:
https://pre-commit.com/#installation

# run framework manually in git repo
pre-commit run -a

# install git hooks permanently for local checkout
pre-commit install
```

File: charts/uptime-kuma/templates/ingress.yaml [hunk 1]
```diff
@@ -54,10 +53,10 @@ spec:
               service:
                 name: {{ $fullName }}
                 port:
-                  number: {{ $svcPort }}
+                  name: http
               {{- else }}
               serviceName: {{ $fullName }}
-              servicePort: {{ $svcPort }}
+              servicePort: {{ $.Values.service.port }}
```
Comment (dirsigler) [inline charts/uptime-kuma/templates/ingress.yaml:59]: I think the `$` at the beginning of `$.Values.service.port` is not needed here.

**Review by dirsigler — CHANGES_REQUESTED** [review]
added question.

**Comment by marcules** [comment]
Hi @dirsigler I'll look into this in a couple of days:

1. your change req. is valid, it is not strictly necessary in that case - but it also "doesn't hurt" - I'll change it nevertheless
2. the changes to upstrea uptime-kuma were not yet released in a docker image, I'll wait for that to happen and also update the docker image here
3. the override / setting of UPTIME_KUMA_PORT would not be strictly necessary after the fix, but I would leave it in so users can run on a non-standard port, what do you think? It probably is not necessary to set the (pod/container internal) port at all, because the exposed service port is configurable and can map to that standard port; with this change both of them could be configured independently, just not sure if it's needed
4. pod-labels should maybe not be set for all selector labels? The previous configuration was/is consistent between deployment and statefulset, but IMHO "extra labels" and "selector labels" are not the same and should not be conflated - do you think we should keep the extra labels in selector-labels (consistent with previous impl.) - or move them out?
5. Have a nice easter holiday

**Review by dirsigler — COMMENTED** [review]
lgtm

**Review by dirsigler — APPROVED** [review]

**Pull request closed without merging**